### DISCUSSION TITLE
Supporting Library , Datatypes , Querytype and Index types

### DISCUSSION THREAD

**Discussion by pauldintel** [qdrant/qdrant#2910] [General]
hi Team , 
Like other vector DB I am looking what Libraries and Index type from below is supported by Qdrant

HNSWlib
DiskANN / Vamana
ScaNN
FAISS-IVFP
NGT

Also are all these datatypes supported ?
FLOAT
INT8 , UINT8
FLOAT16
BFLOAT16

And does Qdrant has support of Intel AVX instructions like AVX512 ?

What are the query types from below it supports ?

Search
Filter + Search

thanks

**Comment by generall** [comment]
Hi @pauldintel,

> I am looking what Libraries and Index type from below is supported by Qdrant

Qdrant have custom implementation of HNSW.

> Also are all these datatypes supported ?

By default, it uses 32bit float, but can be compressed with scalar quantization into INT8 and even to bit level with binary quantization.

> And does Qdrant has support of Intel AVX instructions like AVX512 ?

Yes, we support AVX, as well as Neon instruction set (analog of AVX on ARM processors).
AVX512 we tried, but it turned out to be less efficient than AVX, cause of register size and loading time

> What are the query types from below it supports ?

Search - yes
Filter + Search - yes
+
Recommendation request - multiple positive and negative vectors as an example
+ (soon TBA)
discovery requests
diversity requests

**Comment by pauldintel** [comment]
thank you @generall  for good details ... so only custom HNSW is supported and no other algorithms ? 
Are you saying : "scalar quantization into INT8" is available in today's code ? or WIP .

**Comment by generall** [comment]
we decided that it is better to optimize one algorithm than have rudimentary support to many.

Regarding scalar quantization - yes, it is already available

**Reply by LuoTengBin** [reply]
Hi @generall !
Is QDRANT's SIMD hardware acceleration enabled by default, or does it require manual configuration to enable?

**Reply by generall** [reply]
it's enabled by default

**Reply by LuoTengBin** [reply]
Thank you for your explanation.
